crates/rattler_networking/src/s3_middleware.rs

```diff
@@ -80,15 +84,30 @@ impl S3 {
             config,
             auth_storage,
             expiration: std::time::Duration::from_secs(300),
+            default_client: Arc::new(OnceCell::new()),
+            provider_name: "rattler",
         }
     }
+
+    /// Sets the provider name that is used by the s3 client.
```

Comment:
maybe add a small comment on what this does? it's not immediate to me what this is used for

Reply:
I think it was only used to identify where certain credentials come from. I hardcoded this from rattler and removed the ability to configure it.